Doppler.BillingUser/Controllers/BillingController.cs

```diff
@@ -843,11 +873,30 @@ private async Task<int> BuyCredits(UserTypePlanInformation currentPlan, UserType
                 await _promotionRepository.IncrementUsedTimes(promotion);
             }
 
+            var status = !isPaymentPending ? PaymentStatusEnum.Approved.ToDescription() : PaymentStatusEnum.Pending.ToDescription();
+            CreateUserPaymentHistory(user.IdUser, (int)user.PaymentMethod, agreementInformation.PlanId, status, billingCreditId);
+
             //Send notifications
             SendNotifications(user.Email, newPlan, user, partialBalance, promotion, agreementInformation.Promocode, agreementInformation.DiscountId, payment, billingCreditType, currentPlan, null);
 
-
             return billingCreditId;
         }
+
+        private async void CreateUserPaymentHistory(int idUser, int idPaymentMethod, int idPlan, string status, int idBillingCredit)
```

Comment:
Cuidado! los [async void son peligrosos](https://learn.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#avoid-async-void)

```suggestion
        private async Task CreateUserPaymentHistory(int idUser, int idPaymentMethod, int idPlan, string status, int idBillingCredit)
```